obs_maven/repo.py

```diff
@@ -66,7 +77,7 @@ def get_binary(self, path, target, mtime):
         """
         Equivalent of osc.core.get_binary_file
         """
-        f = urllib.request.urlopen("{}/{}/{}/{}".format(self.uri, self.project, self.repository, path))
+        f = urllib.request.urlopen(self.get_repo_path(path))
```

Comment:
Bug!

You should be really dropping legacy `urllib`. Python3 in docs recommends `requests` instead. So first, you should `try/except` that, as it raises [URLError](https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError). Another bug that you must also check `f` variable if it is not None, otherwise it will crash at runtime.

Reply:
If I may suggest, I would rather still dump that `urllib` if I were you. The `requests` are way better, simpler, safer and you will love it next moment. :wink: